spec/lib/api/2.0/users-spec.js

```diff
@@ -298,4 +298,53 @@ describe('Http.Api.Users', function () {
                     .expect(401);
             });
     });
+
+    it('should 400 a bad username post without access', function() {
+        accountService.listUsers.resolves([ userObj ]);
+        return helper.request().post('/login')
+            .send({username: " readonly", password: "read123"})
+            .expect(201)
+            .then(function(res) {
+                return res.body.token;
+            }).then(function(token) {
+                return helper.request().post('/api/2.0/users')
+                    .set("authorization", 'JWT ' + token)
+                    .send(readOnlyObj)
```

Comment:
the bad user name is set in line 305, so the 400 expectation should be returned earlier in line 306. Actually the `readOnlyObj` has the valid username, so it's returned http status should be success.

From your test purpose, the bad username need be set in line 312, the username (line 305) for `/login` should be valid.